the_debuginator.h

```diff
@@ -2377,6 +2377,30 @@ void debuginator_move_to_parent(TheDebuginator* debuginator) {
 	}
 }
 
+void debuginator_move_to_root(TheDebuginator* debuginator) {
+	DebuginatorItem* hot_item = debuginator->hot_item;
+	while (hot_item && hot_item != debuginator->root) {
```

Comment:
I feel like this is overly complex, though there could be some case I'm missing. Basically what you're doing it setting the hot item to the root's first item, and you're ensuring that if the current hot item is an expanded leaf, you close it first. 

I don't think you need a while loop for that...